### PR TITLE
Allow Multi-Version of ScaleIO with GoScaleIO API Bindings

### PR DESCRIPTION
**Pull request by oskoss**
Added new Version parameter to the configConnect struct. Most changes occurred to the HTTP headers.

### CODE REVIEW

**Comment by clintkitson** [comment]
@oskoss Thanks for the contribution! I believe there is a /version can be used to determine this dynamically. That would be a great addition to this functionality to ensure seamless support of both versions. Can you adapt this to do a version lookup and use that version later on?

File: api.go [hunk 1]
```diff
@@ -20,11 +20,12 @@ import (
...
-	Insecure      string
-	ShowBody      bool
+	Token       string
+	SIOEndpoint url.URL
+	Http        http.Client
+	Insecure    string
+	ShowBody    bool
+
 	configConnect *ConfigConnect
 }
```
Comment (koensayr) [inline api.go:30]: This looks like just an issue of Whitespace, can we fix this so we don't merge a whitespace diff?

**Comment by oskoss** [comment]
Okay implemented both requests. How's it looking now?

File: api.go (hunk 1 above)
Comment (akutz) [inline api.go:30]: Hi @koensayr,

The issue is that the original author (_cough_ @clintonskitson _cough_) didn't use `gofmt` to format the sources, and now he does and we all do, and thus the change you see above. This should happen far less frequently as all our code uses `gofmt` by default per the build process not to mention Atom has it built in.

**Comment by clintkitson** [comment]
@Oskoss Better thank you @oskoss. Will be testing this further and will let you know.

**Comment by akutz** [comment]
Hi @clintonskitson,

We're in a hangout right now. I just told him that since the current test suite doesn't pass (and he says it fails even without this) that you'll need to approve the merge. Thanks!

**Comment by clintkitson** [comment]
@oskoss Just had a chance to review this in more detail. I am debating whether the right place to put the version is in an exported/unexported value and also whether or not to put in directly in the Client object.

In order to enable someone to force a version manually, I believe the proper type would be an exported field. Also this would dictate that before setting the version that a check was made to see if someone explicitly configured the version. Thoughts?

**Comment by oskoss** [comment]
@clinton at first I was on the side of allowing the version be set with an exported values, thinking that the user could be running two (different scaleio versions) gateways on a single host. Thinking more about this and implementing the /version piece of the PR I realized that would never be possible since the port that the gateway runs on is unconfigurable, and therefore you cannot run any more instances of the gateway than one. Thusly the /version check is all we need, no exported variables at all.

**Comment by kacole2** [comment]
We may want to think about making the v2 API the default and having some flag or ENV for the older API. I would assume 90% of the use case moving forward will be the v2 API.

**Comment by clintkitson** [comment]
Playing this forward a bit, at some future point the bindings should be abstracted enough to handle major differences in the API versions. At this point, the version coming back from the API may be one source of indication that we need to invoke the v2/v3 part of the package to support the API. Today, with a simple pass through as we are doing this isn't as important.

Being able to set the API version manually per a flag is important since there may be situations where the API bindings are set for a specific version, and the ScaleIO API has the ability to be backwards compatible. The version that is sent to ScaleIO at that point would a) either trigger a "I do not support this rev" or "ok, I will talk the version that you are compatible with".

We have an option to hardcode the version into the API bindings, but I don't like this option (hence why we are just looking up and resending that version on today). Based on the complexity how we are using the API today with RR, we haven't encountered any breaking changes to the API as is. So in order to maintain the most flexibility moving forward, I think it's a good idea to enable a parameter in RR that sets the version which would then override the "automatic lookup" that @oskoss added. Other than that a auto-detect and pass on works until we implement the right abstraction to break out calls into separate v1/v2/v3 packages/drivers.

**Comment by oskoss** [comment]
@clintonskitson understandable to me, for now I would keep this PR the same like you are suggesting. If you would rather, I can also change the PR to look for the env and if not found query the /version endpoint and use that as its version. Also we had a PR against Rex Ray but closed it since this pull request doesnt require a env anymore at all. I can reopen if needed.

**Comment by clintkitson** [comment]
The suggestion was to include a Version field that can be set as part of
the New.. method. This field should be checked before doing the checkver.

I have a PR for RR that points to your branch until we merge your goscaleio
PR. I will take care of adding the config option to RR to manually specify
version as an optional field if u could get the version param working in
goscaleio.

**Comment by oskoss** [comment]
@clintonskitson Okay do you mind checking out the PR now, I believe it is good to go. Also let me know if you are working on the PR for RR since NewClientWithArgs now has an additional field, if not I can add a commit to your PR over on RR.

**Pull request merged**